File: examples/typegraphs/basic.ts
// skip:start
import { Policy, t, typegraph } from "@typegraph/sdk";
import { Auth } from "@typegraph/sdk/params";
import { DenoRuntime } from "@typegraph/sdk/runtimes/deno";

// skip:end

await typegraph(
  {
    name: "basic-authentication",
    // skip:next-line
    cors: { allowOrigin: ["https://metatype.dev", "http://localhost:3000"] },
  },
  (g) => {
    const deno = new DenoRuntime();
    const pub = Policy.public();

    const ctx = t.struct({
      username: t.string().optional(),
    });

    // highlight-next-line
    g.auth(Auth.basic(["admin"]));

    g.expose({
      get_context: deno
        .identity(ctx)
        .apply({
          username: g.fromContext("username"),
        })
        .withPolicy(pub),
    });
  },
);

File: examples/typegraphs/basic.py
# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
# SPDX-License-Identifier: MPL-2.0

# skip:start
from typegraph import Graph, Policy, t, typegraph
from typegraph.graph.params import Auth, Cors
from typegraph.runtimes.deno import DenoRuntime

# skip:end


@typegraph(
    # skip:start
    cors=Cors(allow_origin=["https://metatype.dev", "http://localhost:3000"]),
    # skip:end
)
def basic_authentication(g: Graph):
    deno = DenoRuntime()
    public = Policy.public()

    ctx = t.struct({"username": t.string().optional()})

    # highlight-next-line
    g.auth(Auth.basic(["admin"]))

    g.expose(
        public,
        get_context=deno.identity(ctx).apply(
            {
                "username": g.from_context("username"),
            },
        ),
    )
